subsys/mpsl/cx/Kconfig

```diff
@@ -4,8 +4,20 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+DT_COMPAT_CX_NRF700X := nordic,nrf700x-coex
+# config MPSL_CX_SUPPORTED
+
+config MPSL_CX_ANY_SUPPORT
+	bool
+	default $(dt_nodelabel_enabled,nrf_radio_coex)
+
+config MPSL_CX_NRF700X_SUPPORT
+	bool
+	default $(dt_nodelabel_has_compat,nrf_radio_coex,$(DT_COMPAT_CX_NRF700X))
+
 config MPSL_CX
 	bool "Radio Coexistence interface support"
+	default MPSL_CX_ANY_SUPPORT && WIFI && (BT || NRF_802154_RADIO_DRIVER || NRF_802154_SER_HOST)
```

Comment:
The idea here is that it is the general `CX`, which can also be used with other WiFi chips and PTAs. So, if the user uses any WiFi driver (for example a proprietary one), so that `CONFIG_WIFI` is set to `y` and he provides any `nrf_radio_coex` mode, the assumption is that by default MPSL_CX should be enabled. So it this stage, it doesn't matter that it is the nRF700x Wifi driver.